core/src/price.ts

```diff
@@ -49,3 +49,39 @@ export const calculateTransactionProfit = function (auction: Auction): BigNumber
     const totalMarketPriceLimitedByDebt = collateralAmountLimitedByDebt.multipliedBy(auction.marketUnitPrice);
     return totalMarketPriceLimitedByDebt.minus(auction.debtDAI);
 };
+
+export const calculateTransactionProfitDate = function (auction: Auction): Date | undefined {
+    if (
+        auction.secondsBetweenPriceDrops === undefined ||
+        auction.secondsTillNextPriceDrop === undefined ||
+        auction.marketUnitPrice === undefined ||
+        auction.priceDropRatio === undefined
+    ) {
+        return undefined;
+    }
+
+    const isAlreadyProfitable = auction.approximateUnitPrice.isLessThan(auction.marketUnitPrice);
+    const date = new Date();
+
+    let steps = 0;
+    let currentValue = auction.approximateUnitPrice;
+    let secondsTillProfitable = 0;
+
+    if (isAlreadyProfitable) {
+        while (currentValue.isLessThan(auction.marketUnitPrice)) {
+            steps -= 1;
+            currentValue = currentValue.multipliedBy(new BigNumber(1).minus(auction.priceDropRatio).plus(1));
+        }
+        secondsTillProfitable = auction.secondsBetweenPriceDrops * steps + auction.secondsTillNextPriceDrop;
```

Comment:
If the auction is already profitable, the profitability indicator is no longer viable information

Reply:
Ah, true. I was also wondering how would you solve this proposal 🙂
Maybe you just return some random date in the past instead of undefined or just undefined + handle it in the component.

Reply:
My main suggestion here: we don't have to display exact time if the auction is already profitable and therefore we don't have to calculate it.

Reply:
> Maybe you just return some random date in the past instead of undefined 😅

Was thinking this too, but I guess this would be very confusing when looking at this function alone 😅

> But yeah, maybe undefined + handling this it in the component would be a better solution

Doing this now 👍